cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -871,6 +883,11 @@ private NamespaceId validateNamespace(String namespaceId) throws BadRequestExcep
     }
     return namespace;
   }
+  
+  private String getLatestAppVersion(ApplicationId applicationId) throws ApplicationNotFoundException, IOException {
+    ApplicationDetail latestApp = applicationLifecycleService.getLatestAppDetail(applicationId);
```

Comment:
Deserializing the whole App details and just to get the version ID is expensive. I think it's better to have a `List<ApplicationId> getLatestAppVersions(NamespaceId namespace, List<String> appNames)` method in the app lifecycle service.

Reply:
@de-lan is making this change now

Reply:
changed pushed